c2corg_ui/static/js/searchfilters.js

```diff
@@ -61,76 +52,121 @@ app.module.directive('appSearchFilters', app.searchFiltersDirective);
 
 /**
  * @param {angular.Scope} $scope Scope.
- * @param {angular.JQLite} $element Element.
- * @param {angular.Attributes} $attrs Attributes.
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @param {ngeo.Debounce} ngeoDebounce ngeo Debounce service.
  * @constructor
  * @ngInject
  * @export
  */
-app.SearchFiltersController = function($scope, $element, $attrs) {
-
+app.SearchFiltersController = function($scope, ngeoLocation, ngeoDebounce) {
 
   /**
    * @type {angular.Scope}
    * @private
    */
   this.scope_ = $scope;
 
-  // FIXME: not generic to all document types
-  // Init filters object
   /**
-   * @type {Array<string>}
-   * @export
+   * @type {ngeo.Location}
+   * @private
    */
-  this.activities = [];
-
+  this.location_ = ngeoLocation;
 
   /**
-   * @type {Array<string>}
+   * @type {Object}
    * @export
    */
-  this.waypoint_types = [];
-
+  this.filters = {};
 
   /**
-   * @type {string}
-   * @export
+   * @type {boolean}
+   * @private
    */
-  this.lang;
+  this.loading_ = true;
+
+  // Fill the filters according to the loaded URL parameters
+  var keys = this.location_.getParamKeys().filter(function(x) {
+    return app.SearchFiltersController.IGNORED_FILTERS.indexOf(x) === -1;
+  });
+  for (var i = 0, n = keys.length; i < n; i++) {
+    this.getFilterFromPermalink_(keys[i]);
+  }
 
+  // Deep watch is used here because we need to watch the list filters as well
+  // which a simple $watch or $watchCollection does not. Might cause
+  // perf/memory issues though...
+  this.scope_.$watch(function() {
+    return this.filters;
+  }.bind(this), ngeoDebounce(
+      this.handleFiltersChange_.bind(this),
+      500, /* invokeApply */ true), /* deep watch */ true
+  );
+};
 
-  /**
-   * @type {string}
-   * @export
-   */
-  this.around;
 
+/**
+ * @const
+ * @type {Array.<string>}
+ */
+app.SearchFiltersController.IGNORED_FILTERS = ['bbox', 'offset', 'limit'];
 
-  /**
-   * @type {Object}
-   * @export
-   */
-  this.height_diff_up = {min: 0, max: 9999};
 
-  /**
-   * @type {Object}
-   * @export
-   */
-  this.elevation = {min : 0, max: 9999};
+/**
+ * @param {string} key Filter key.
+ * @private
+ */
+app.SearchFiltersController.prototype.getFilterFromPermalink_ = function(key) {
+  // TODO find a more generic way or at least factorize code
```

Comment:
OK I have just pushed some refactoring of this code (d448c98), a bit like what you have done in the API.
My main concern is that I have to explicitely test for some specific parameters (eg. "walt", "wtyp") to convert them using the right filters creating functions.
I have tried to pass the filter types in the filter forms themselves using for instance
```
ng-init="filtersCtrl.filterTypes.wtyp = 'list'"
```
but it's a bit complicated making sure the types info are actually loaded when running the permalink parsing.

Anyway I prefer postponing this issue to a future "advanced search step2" issue/PR.